pkg/git/gitproviderfactory/gitproviderfactory.go

```diff
@@ -50,28 +49,50 @@ type GitClientConfig struct {
 // createGitClient creates new git provider client for the requested config
 func createGitClient(gitClientConfig GitClientConfig) (gitprovider.GitProviderClient, error) {
 	gitProvider := gitClientConfig.GitProvider
-	config := gitClientConfig.PacSecretData
+	secretData := gitClientConfig.PacSecretData
+	username, usernameExists := secretData["username"]
+	password, passwordExists := secretData["password"]
+	_, sshKeyExists := secretData["ssh-privatekey"]
 
-	isAppUsed := gitops.IsPaCApplicationConfigured(gitProvider, config)
-	var accessToken string
-	if !isAppUsed {
-		accessToken = strings.TrimSpace(string(config[gitops.GetProviderTokenKey(gitProvider)]))
-	}
+	isAppUsed := gitops.IsPaCApplicationConfigured(gitProvider, secretData)
 
 	switch gitProvider {
 	case "github":
 		if !isAppUsed {
-			return github.NewGithubClient(accessToken), nil
+			if usernameExists && passwordExists {
```

Comment:
I think we can have `if passwordExists {` and then analyze username. That would remove duplication on password decoding. But up to you.